src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/RegexNodeToSymbolicConverter.cs

```diff
@@ -5,326 +5,194 @@
-                // performed above.
-                if (subtractorCond is not null)
-                {
-                    moveCond = Solver.And(moveCond, Solver.Not(subtractorCond));
-                }
-
-                return moveCond;
-
-                static List<(char First, char Last)> ComputeRanges(string set)
-                {
-                    int setLength = set[RegexCharClass.SetLengthIndex];
-
-                    var ranges = new List<(char, char)>(setLength);
-                    int i = RegexCharClass.SetStartIndex;
-                    int end = i + setLength;
-                    while (i < end)
+                case RegexNodeKind.Alternate:
                     {
-                        char first = set[i];
-                        i++;
-
-                        char last = i < end ?
-                            (char)(set[i] - 1) :
-                            RegexCharClass.LastChar;
-                        i++;
-
-                        ranges.Add((first, last));
+                        // Alternations are created by creating an Or of all of its children.
+                        // This Or needs to be "ordered" to achieve the same semantics as the backtracking engines.
+                        var branches = new SymbolicRegexNode<BDD>[node.ChildCount()];
+                        for (int i = 0; i < branches.Length; i++)
+                        {
+                            branches[i] = ConvertToSymbolicRegexNode(node.Child(i), tryCreateFixedLengthMarker);
+                        }
+                        return _builder.OrderedOr(branches);
                     }
 
-                    return ranges;
-                }
+                // Loops
 
-                BDD MapCategoryCodeSetToCondition(HashSet<int> catCodes)
-                {
-                    // TBD: perhaps other common cases should be specialized similarly
-                    // check first if all word character category combinations are covered
-                    // which is the most common case, then use the combined predicate \w
-                    // rather than a disjunction of the component category predicates
-                    // the word character class \w covers categories 0,1,2,3,4,8,18
-                    BDD? catCond = null;
-                    if (catCodes.Contains(0) && catCodes.Contains(1) && catCodes.Contains(2) && catCodes.Contains(3) &&
-                        catCodes.Contains(4) && catCodes.Contains(8) && catCodes.Contains(18))
+                case RegexNodeKind.Oneloop:
+                case RegexNodeKind.Onelazy:
+                case RegexNodeKind.Notoneloop:
+                case RegexNodeKind.Notonelazy:
                     {
-                        catCodes.Remove(0);
-                        catCodes.Remove(1);
-                        catCodes.Remove(2);
-                        catCodes.Remove(3);
-                        catCodes.Remove(4);
-                        catCodes.Remove(8);
```

Comment:
@olsaarik, this was a bug that it was missing a Contains/Remove for 5, yes?

Reply:
Yes, it is indeed a bug, because \w is the union of the 8 categories: 0,1,2,3,4,5,8,18